Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -124,5 +334,113 @@ namespace AZ
 
             return mipChainAsset->GetSubImageData(mip - mipChain.m_mipOffset, slice);
         }
+
+        template<typename T>
+        T StreamingImageAsset::GetSubImagePixelValueInternal(uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            AZStd::array<T, 1> values = { aznumeric_cast<T>(0) };
+
+            auto position = AZStd::make_pair(x, y);
+            AZStd::span<T> valueSpan(values.begin(), values.size());
+            GetSubImagePixelValues(position, position, valueSpan, componentIndex, mip, slice);
+
+            return values[0];
+        }
+
+        template<>
+        float StreamingImageAsset::GetSubImagePixelValue<float>(uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            return GetSubImagePixelValueInternal<float>(x, y, componentIndex, mip, slice);
+        }
+
+        template<>
+        AZ::u32 StreamingImageAsset::GetSubImagePixelValue<AZ::u32>(uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            return GetSubImagePixelValueInternal<AZ::u32>(x, y, componentIndex, mip, slice);
+        }
+
+        template<>
+        AZ::s32 StreamingImageAsset::GetSubImagePixelValue<AZ::s32>(uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            return GetSubImagePixelValueInternal<AZ::s32>(x, y, componentIndex, mip, slice);
+        }
+
+        void StreamingImageAsset::GetSubImagePixelValues(AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<float> outValues, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            // TODO: Use the component index
+            (void)componentIndex;
+
+            auto imageData = GetSubImageData(mip, slice);
+
+            if (!imageData.empty())
+            {
+                const AZ::RHI::ImageDescriptor imageDescriptor = GetImageDescriptor();
+                auto width = imageDescriptor.m_size.m_width;
+
+                size_t outValuesIndex = 0;
+                for (uint32_t y = topLeft.second; y <= bottomRight.second; ++y)
+                {
+                    for (uint32_t x = topLeft.first; x <= bottomRight.first; ++x)
+                    {
+                        size_t imageDataIndex = (y * width) + x;
+
+                        auto& outValue = const_cast<float&>(outValues[outValuesIndex++]);
```

Comment:
Repeated: The const_cast shouldn't be needed any more, now that we're using span<>

Reply:
Perfect, thanks will switch all these!